biz.aQute.remote/src/aQute/remote/util/AgentSupervisor.java

```diff
@@ -41,15 +42,26 @@
 	}
 
 	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port) throws Exception {
-		while (true)
+		connect(agent, supervisor, host, port, -1);
+	}
+
+	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, int timeout)
+			throws Exception {
+		while (timeout == -1 || timeout > 0)
```

Comment:
I think you can leave this as a `while (true)` loop. If successful, it always returns from the try block. The same should be done from the catch block: If the time out is expired, then return. Otherwise loop back up. This avoids the awkward test in the while statement.